### PR TITLE
[SYCL][UR] Simplify platform binding

### PR DESCRIPTION
**Pull request by jandres742**

### CODE REVIEW

**Comment by jandres742** [comment]
@smaslov-intel : please review

File: sycl/plugins/level_zero/pi_level_zero.cpp [hunk 1]
```diff
@@ -208,7 +208,9 @@ static const std::pair<int, int> getRangeOfAllowedComputeEngines() {
   return std::pair<int, int>(0, INT_MAX);
 }
 
-pi_platform _pi_context::getPlatform() const { return Devices[0]->Platform; }
+pi_platform _pi_context::getPlatform() const {
+  return reinterpret_cast<pi_platform>(Devices[0]->Platform);
```
[the remark this answers is not in the document]
Comment (jandres742) [reply sycl/plugins/level_zero/pi_level_zero.cpp:212]: @smaslov-intel : the idea is to not need them, correct. See here for the final patch https://github.com/jandres742/llvm/blob/bff8057c16d195c9a34035423c2eafe5eb093b51/sycl/plugins/unified_runtime/ur/adapters/level_zero/ur_level_zero_context.cpp#L339, where no cast is needed. So this is for now while we move the code to the UR L0 adapter.

[the remark this answers is not in the document]
Comment (smaslov-intel) [reply sycl/plugins/level_zero/pi_level_zero.cpp:212]: Yeah, I understand that when everything is in UR L0 there will not be a cast to PI.
Just maybe for still remaining PI L0 codes not need a cast, keep the bindings?
You will not need those pi_level_zero.cpp changes

**Comment by jandres742** [comment]
Moving to draft, while I try other approach. Thanks @smaslov-intel for the comments.

**Pull request closed without merging**